spec/requests/submissions_search_spec.rb

```diff
@@ -640,5 +640,153 @@
         expect(laa_references).to match(%w[LAA-CCCCCC LAA-BBBBBB LAA-AAAAAA])
       end
     end
+
+    context "when searching for queries that may be invalid" do
+      before do
+        create(:submission,
+               :with_pa_version,
+               laa_reference: "LAA-AAAAAA",
+               defendant_name: "Fred Arbor",
+               firm_name: "Aardvark & Co")
+
+        create(:submission,
+               :with_pa_version,
+               laa_reference: "LAA-BBBBBB",
+               defendant_name: "Fred Bloggs",
+               firm_name: "Smith & (Partners) Ltd.")
+
+        create(:submission,
+               :with_pa_version,
+               laa_reference: "LAA-CCC123",
+               defendant_name: "Jimmy Buffer",
+               firm_name: "Über Legal Co.",
+               ufn: "311223/001")
+
+        create(:submission,
+               :with_pa_version,
+               laa_reference: "LAA-MiXeD1",
+               defendant_name: "Test Person",
+               firm_name: "Legal Eagles",
+               ufn: "123456")
+
+        create(:submission,
+               :with_pa_version,
+               laa_reference: "LAA-PUNC28",
+               defendant_name: "James O'Connor-Smith",
+               firm_name: "Legal & Law (International) Ltd.")
+      end
+
+      it "handles a completely invalid query" do
+        post search_endpoint, params: { query: "." }
+
+        expect(response).to have_http_status(:created)
+        expect(response.parsed_body["data"]).to match([])
+      end
+
+      it "ignores unmatched parentheses" do
+        post search_endpoint, params: { query: "Fred)" }
+
+        expect(response).to have_http_status(:created)
+        expect(response.parsed_body["data"].pluck("client_name")).to match(["Fred Bloggs", "Fred Arbor"])
+      end
+
+      it "ignores a real query attempt" do
+        post search_endpoint, params: { query: "LAA-AAAAAA)" }
+
+        expect(response).to have_http_status(:created)
+        expect(response.parsed_body["data"].pluck("laa_reference")).to match(%w[LAA-AAAAAA])
+      end
+
+      it "handles mixed case references" do
+        post search_endpoint, params: { query: "LAA-MiXeD1" }
+
+        expect(response).to have_http_status(:created)
+        expect(response.parsed_body["data"].pluck("laa_reference")).to match(%w[LAA-MiXeD1])
+      end
+
+      it "handles complex names" do
+        post search_endpoint, params: { query: "O'Connor-Smith" }
+
+        expect(response).to have_http_status(:created)
+        expect(response.parsed_body["data"].pluck("client_name")).to eq(["James O'Connor-Smith"])
+      end
+
+      it "ignores unescaped ampersands" do
```

Comment:
this doesn't really check if it ignores the ampersand. I think you need another firm name like `aardvark smithson` in your dataset. That will only be returned if the & is ignored.

Reply:
Unescaped in the query, which is what one of the issues was; having an `&` in the query broke the search.

Reply:
That makes sense, have amended tests to hopefully demonstrate what's happening.